autopilot/autopilot.go

```diff
@@ -150,11 +157,35 @@ func (ap *Autopilot) Run() error {
 func (ap *Autopilot) Stop() error {
 	ap.ticker.Stop()
 	close(ap.stopChan)
+	ap.wg.Wait()
 	return nil
 }
 
+func (ap *Autopilot) actionsHandler(jc jape.Context) {
+	var since time.Time
+	max := -1
+	if jc.DecodeForm("since", (*paramTime)(&since)) != nil || jc.DecodeForm("max", &max) != nil {
+		return
+	}
+	jc.Encode(ap.Actions(since, max))
+}
+
+func (ap *Autopilot) configHandlerGET(jc jape.Context) {
+	jc.Encode(ap.Config())
+}
+
+func (ap *Autopilot) configHandlerPUT(jc jape.Context) {
+	var c Config
+	if jc.Decode(&c) != nil {
+		return
+	}
+	if jc.Check("failed to set config", ap.SetConfig(c)) != nil {
+		return
+	}
+}
+
 // New initializes an Autopilot.
-func New(store Store, bus Bus, worker Worker, logger *zap.Logger, heartbeat time.Duration) (*Autopilot, error) {
+func New(store Store, bus Bus, worker Worker, logger *zap.Logger, heartbeat time.Duration) (_ http.Handler, run, cleanup func() error) {
```

Comment:
I did this for the `worker` and `bus`, but I don't think it's appropriate for the `autopilot`. The autopilot is a big stateful actor with goroutines and complicated shutdown etc. etc. There's no way to avoid it.